AlphaWallet/Core/Types/RateLimiter.swift

```diff
@@ -46,3 +46,62 @@ class RateLimiter {
         }
     }
 }
+
+public protocol SyncLimiter {
+    @discardableResult func execute(_ block: () -> Void) -> Bool
+    func reset()
+}
+
+extension SyncLimiter {
+    public func execute<T>(_ block: () -> T) -> T? {
+        var value: T? = nil
+
+        execute {
+            value = block()
+        }
+
+        return value
+    }
+}
+
+public final class TimedLimiter: SyncLimiter {
+
+    // MARK: - Properties
+    public let limit: TimeInterval
+    public private(set) var lastExecutedAt: Date?
+
+    private let syncQueue = DispatchQueue(label: "com.samsoffes.ratelimit", attributes: [])
```

Comment:
Remember to change the label if we keep this :)

Reply:
yep it acts in same way as RateLimiter, but this one avoids using timer creating, (and that is more expensive operation).

Reply:
I see. If it does the same thing but the implementation is better, would you replace the internals of RateLimiter in a separate commit/PR and use it still? No point having 2 classes if they do the same thing.